app/src/main/java/dev/leonlatsch/photok/settings/domain/GetVaultStatsUseCase.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ *   Copyright 2020-2024 Leon Latsch
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package dev.leonlatsch.photok.settings.domain
+
+import dev.leonlatsch.photok.model.database.dao.PhotoDao
+import javax.inject.Inject
+
+/**
+ * Data class representing vault statistics.
+ */
+data class VaultStats(
+    val totalFiles: Int,
+    val totalSizeBytes: Long,
+)
+
+/**
+ * Use case to retrieve vault statistics.
+ *
+ * @since 2.0.0
+ */
+class GetVaultStatsUseCase @Inject constructor(
```

Comment:
Please make these return a flow.

You can use combine and combine 2 flows from the dao.

Also a clean arch thing: Please don't use dao in a use case. Use the repository interface. I think PhotoRepository has no interface yet, but there could be a VaultRepository interface with a VaultRepositoryImpl using photoDao.

You can orient on AlbumRepository.